### PR TITLE
angular: angular.forEach iterator should have obj parameter

### PR DESCRIPTION
**Pull request by vetvicka**
If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.angularjs.org/api/ng/function/angular.forEach

> The iterator function is invoked with iterator(value, key, obj), where value is the value of an object property or an array element, key is the object property key or array element index and obj is the obj itself.

### CODE REVIEW

File: angular/index.d.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ declare namespace angular {
          * @param iterator Iterator function.
          * @param context Object to become context (this) for the iterator function.
          */
-        forEach<T>(obj: T[], iterator: (value: T, key: number) => any, context?: any): any;
+        forEach<T>(obj: T[], iterator: (value: T, key: number, obj?: T[]) => any, context?: any): any;
```
Comment (thorn0) [inline angular/index.d.ts]: It should be `obj`, not `obj?`. See http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks
Also please change the return types for both the callback (should be `void`) and `forEach` (it returns `obj`).

Comment (vetvicka) [inline angular/index.d.ts]: I have also changed return type of `forEach`. It should return reference to `obj`.

**Comment by thorn0** [comment]
I'm not sure what's going on with the tests. Can you please try rebasing it on top of master?

**Comment by thorn0** [comment]
Oh, that's not how 'rebasing it on top of master' is supposed look. =)

**Comment by vetvicka** [comment]
oops, my bad :(
I'm not sure what to do with this branch. I'll make new pull request next week.

**Comment by thorn0** [comment]
If you use the same name (`foreach`) for the new branch and `push --force` it, this PR will see it.

**Pull request closed without merging**